### PR TITLE
Http url string obfuscation

### PR DESCRIPTION
**Pull request by thedavl**
# What does this PR do?
 
Adds an HTTP url string obfuscator, does a small refactor to mini agent configuration

# Motivation

Mini Agent featureset

# Additional Notes

Anything else we should know when reviewing?

# How to test the change?

Unit tests

### CODE REVIEW

File: trace-obfuscation/src/http.rs [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        [
+            test_name           [remove_path_digits_9]
+            remove_query_string [false]
+            remove_path_digits  [true]
+            input               ["http://user:password@foo.com/1/2/3?q=james"]
+            expected_output     ["http://foo.com/?/?/??q=james"];
+        ]
+    )]
+    #[test]
+    fn test_name() {
```
Comment (thedavl) [inline trace-obfuscation/src/http.rs:170]: test cases were taken from the go agent http obfuscation unit tests

File: trace-obfuscation/src/obfuscation_config.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+#[derive(Debug)]
+pub struct ObfuscationConfig {
+    pub tag_replace_rules: Option<Vec<ReplaceRule>>,
+    pub http_remove_query_string: bool,
+    pub http_remove_path_digits: bool,
+}
+
+impl ObfuscationConfig {
+    pub fn new() -> Result<ObfuscationConfig, Box<dyn std::error::Error>> {
+        let tag_replace_rules: Option<Vec<ReplaceRule>> = match env::var("DD_APM_REPLACE_TAGS") {
```
Comment (maxday) [inline trace-obfuscation/src/obfuscation_config.rs:20]: are we supporting config from datadog.yaml file?

Comment (thedavl) [reply trace-obfuscation/src/obfuscation_config.rs:20]: not currently, no. just from env vars

**Review by maxday — APPROVED** [review]
LGTM! good testing 🔥

**Pull request merged**